tests/platform/test_platform.py

```diff
@@ -163,14 +192,57 @@ def get_anneal_qprogram(runcard, flux_to_bus_topology):
     readout_waveform = IQPair(I=r_wf_I, Q=r_wf_Q)
     weights_shape = Square(amplitude=1, duration=readout_duration)
     weights = IQPair(I=weights_shape, Q=weights_shape)
+
+    qp_anneal = QProgram()
+    shots_variable = qp_anneal.variable("num_shots", Domain.Scalar, int)
+    with qp_anneal.for_loop(variable=shots_variable, start=0, stop=num_shots, step=1):
+        with qp_anneal.average(num_averages):
+            for bus, waveform in anneal_waveforms.items():
+                qp_anneal.play(bus=bus, waveform=waveform)
+            qp_anneal.sync()
+            with qp_anneal.block():
+                qp_anneal.measure(bus="readout_bus", waveform=readout_waveform, weights=weights)
+    return qp_anneal
+
+
+@pytest.fixture(name="anneal_qprogram_with_preparation")
+def get_anneal_qprogram_with_preparation(runcard, flux_to_bus_topology):
+    platform = Platform(runcard=runcard)
+    platform.flux_to_bus_topology = flux_to_bus_topology
+    anneal_waveforms = {
+        next(element.bus for element in platform.flux_to_bus_topology if element.flux == "phix_q0"): Arbitrary(
```

Comment:
Yes, the whole dictionary should be more concrete imo too, but since we are still exploring the analog model, and probably many changes will happen, we decided to leave it as simple as possible for now.